src/sql/magic.py

```diff
@@ -577,11 +576,21 @@ def interactive_execute_wrapper(**kwargs):
             StatementError,
         ) as e:
             # Sqlite apparently return all errors as OperationalError :/
-            self._error_handling(e, command.sql)
+            detailed_msg = detail(e)
+            self._error_handling(e, command.sql, detailed_msg)
```

Comment:
this code looks pretty convoluted. do we need recursion here? `_error_handling` is calling `_error_handling`.

Reply:
There isn't recursion in [_error_handling](https://github.com/ploomber/jupysql/blob/8a9582a6f2ef9055479d76621dc0063104f28764/src/sql/magic.py#L235).

Reply:
I'm moving all this code to a `ErrorHandler` class. Can I close this PR and merge with the circular dependency PR since I would need that here also? @edublancas

Reply:
oh yeah. I forgot that we had to do this because of the circular dependency. yes, I'll close it now